pyface/ui/qt4/data_view/data_view_widget.py

```diff
@@ -45,10 +45,56 @@
 }
 
 
+class DataViewTreeView(QTreeView):
+    """ QTreeView subclass that handles drag and drop via DropHandlers. """
+
+    _widget = None
+
+    def dragEnterEvent(self, event):
+        if self._widget is not None:
+            widget = self._widget
+            for drop_handler in widget.drop_handlers:
+                if drop_handler.can_handle_drop(event, self):
+                    event.acceptProposedAction()
+                    return
```

Comment:
Can we separate this out into a separate private method on the class? This code is repeated in all 4 methods and if I understand correctly, what we are doing here is try handling the event and pass it to the super class if we cannot.

Reply:
Done